### DISCUSSION TITLE
"Most starred languages" metric

### DISCUSSION THREAD

**Discussion by spenserblack** [lowlighter/metrics#707] [Ideas]
This idea popped into my head and I thought it could be a fun metric.
The "most used languages" metric says something like "here is what I use the most." The "most starred language" metric would say something more like "here is what I *want* to use the most" or "here is the ecosystem that I'm most interested in."

If "most starred language" is too generic (there might be confusion with the "starred topics" plugin), it might be better to name it something like "most starred repository languages" or "most common languages of starred repositories".

**Comment by lowlighter** [comment]
This seems pretty cool indeed!

I kind of like the "here is the ecosystem that I'm most interested in." seems the best fit as it doesn't suggest overlapping with other plugins

Maybe it won't be able to process all starred repositories because of the query limits though 🤔

**Reply by spenserblack** [reply]
Good point, it would be very dependent on the number of repositories the user has starred (some users star *a lot* of repositories). Maybe get the top `x` repositories ordered by most recently starred (if available from API), where `x` is an input with some sane default number.

**Reply by lowlighter** [reply]
Since GitHub is introducing stars list, I wonder if it would be revelant to create "stars list metrics" instead

It would let users sort their stars by their own categories (e.g. "projects I'd like to work with", "cs lessons", "cool stuff", etc.) and metrics would be able to display revelant languages to each star list instead

Not sure the api is already out for these though, it still seems to be in beta for now

**Reply by spenserblack** [reply]
That's a great idea!

If the star categories are used, there should probably be a way to filter out/explicitly include categories. Personally, I'd only like to list my uncategorized stars, since those are kind of my "general usage" stars. My categorized stars are a bit too specialized to be representative of the ecosystem(s) I'm interested in.

**Comment by lowlighter** [comment]
This feature is currently in preview on main branch though it is currently limited to stars list for now 

Since it uses puppeteer it may be usable with regular stars without triggering rate limiter but need to see if it would require a lot of additional work. It's still more reliable to use api rather than scrapping which may break at any time
